Imperatus.rte/Actors/Shared/Robot.lua

```diff
@@ -4,14 +4,17 @@ function Create(self)
 	self.stopTimer = Timer();
 end
 function Update(self)
-	local walkSpeedScalar = 2.3;
-	local legs = {self.FGLeg, self.BGLeg};
-	for _, leg in pairs(legs) do
-		walkSpeedScalar = walkSpeedScalar - (leg and leg.Frame/leg.FrameCount or 1.1);
-	end
-	self:SetLimbPathSpeed(1, self.walkSpeed * walkSpeedScalar);
-	if self.Jetpack then
-		self.Jetpack.Throttle = self.JetTimeLeft/self.JetTimeTotal - 0.5;
-		self.Jetpack.FlashScale = 1 + self.Jetpack.Throttle;
+	if self.Health > 0 then
+		local walkSpeedScalar = 2.3;
+		local legs = {self.FGLeg, self.BGLeg};
+		for _, leg in pairs(legs) do
+			walkSpeedScalar = walkSpeedScalar - (leg and leg.Frame/leg.FrameCount or 1.1);
+		end
+		self:SetLimbPathSpeed(1, self.walkSpeed * walkSpeedScalar);
+		
+		if self.Head and (self.FGArm or self.BGArm or self.FGLeg or self.BGLeg) then
+			local limbs = {self.Head, self.FGArm, self.BGArm, self.FGLeg, self.BGLeg};
+			self.Health = self.MaxHealth * (1 - self.WoundCount/self:GetGibWoundLimit(true, false, false)) * #limbs/5;
```

Comment:
This confuses me. By my understanding it's making limb loss cause health damage, but
1. Why not just use breakwounds?
2. Bit of an edge case but if something adds a new arm with lua, this means the health will suddenly skyrocket for no discernible reason, right?